docs/recipes/DistributingAuthTokenToAPI.md

```diff
@@ -0,0 +1,128 @@
+---
+title: Distributing Auth Token to APIs
+description: Use token stored in Authentication Store with API Sauce
+tags:
+  - Authentication
+  - Apisauce
+  - Guide
+last_update:
+  author: Ellie Croce
+publish_date: 2023-05-09
+---
+
+# Distributing Auth Token to APISauce
+
+Building off of the Ignite Boilerplate, this recipe will show you how to connect your Mobx State Tree Authentication Store with an Apisauce instance to make authenticating your API requests a breeze.
+
+## Review of API Instance and Auth Store
+
+To start off let's quickly review the boilerplate Auth Store and API Instance.
+
+In `app/services/api/api.ts` we have an example API class with an export of a singleton instance of that class at the end of the file. This api singleton is what we're going to use to update the apisauce configuration on the instance outside of this file.
+
+```tsx
+export const DEFAULT_API_CONFIG: ApiConfig = {
+  url: Config.API_URL,
+  timeout: 10000,
+}
+
+
+export class Api {
+  apisauce: ApisauceInstance
+  config: ApiConfig
+
+  constructor(config: ApiConfig = DEFAULT_API_CONFIG) {
+    this.config = config
+    this.apisauce = create({
+      baseURL: this.config.url,
+      timeout: this.config.timeout,
+      headers: {
+        Accept: "application/json",
+      },
+    })
+  }
+
+  async getEpisodes(): Promise<{ kind: "ok"; episodes: EpisodeSnapshotIn[] } | GeneralApiProblem> {
+    ... // example API call that needs authenticating
+  }
+
+}
+
+// Singleton instance of the API for convenience
+export const api = new Api()
+```
+
+Next in `app/models/AuthenticationStore.ts` we have an action to set the auth token in the store after retrieving it.
+
+```tsx
+import { Instance, SnapshotOut, types } from "mobx-state-tree";
+
+export const AuthenticationStoreModel = types
+  .model("AuthenticationStore")
+  .props({
+    authToken: types.maybe(types.string),
+    authEmail: "",
+  })
+  .views((store) => ({}))
+  .actions((store) => ({
+    setAuthToken(value?: string) {
+      store.authToken = value;
+    },
+  }));
+
+export interface AuthenticationStore
+  extends Instance<typeof AuthenticationStoreModel> {}
+export interface AuthenticationStoreSnapshot
+  extends SnapshotOut<typeof AuthenticationStoreModel> {}
+```
+
+## Distributing Auth Token to API Instance
+
+Once we have an auth token and setAuthToken has been called, we need to hydrate the authentication header on all APIs that require requests to be authenticated. One of the easiest ways to do that is to grab the instance(s) in an action on the authStore and update the apiSauce configuration directly like so:
+
+```tsx
```

Comment:
```suggestion
```tsx
// We will need to import the api from the service
import { api } from "app/services/api"

...

```